### ISSUE TITLE
[Yaml] Bug: Escaping quotes in strings throws Symfony\Component\Yaml\Exception\ParseException

### ISSUE THREAD

**Issue by Jeroeny** [symfony/symfony#39265]
**Symfony version(s) affected**: 5.1.9, 5.2.0 and possibly more

**Description**  
When a config file in a symfony app (eg. `config/services.yaml` contains the following yaml, it will throw an exception:
```yaml
    App\Example:
        arguments:
            $foo: 'bla\'
```

See the reproducer for more.

**How to reproduce**  

https://github.com/Jeroeny/reproduce/tree/yaml

Originally thought to be related to https://github.com/symfony/symfony/issues/39229, but it's supposedly a different issue. The patch for that issue was also tested, but did not fix the bug.

**Comment by mdeherder** [comment]
Problem also with this in `Myspace.fr.yaml `
```
yes:        'Oui'
no:         'Non' 
the:        'le'
title:
    list:  'Liste.'
    maj:   'Mise à Jour.'
    view:  'Détail.'
btn:
    display: 'Afficher la Sélection'
    new:    'Nouvel Enregistrement'
    return: 'Retour...'
```
Un extra space after 'Non'  causes the following error :

`An exception has been thrown during the rendering of a template ("The file "C:\Symfony\apphome/translations\Myspace.fr.yaml" does not contain valid YAML: Unexpected characters near "ouvel Enregistrement'" at line 18 (near " new: 'Nouvel Enregistrement'").").`

**Comment by mediproductions** [comment]
Same error on 4.4.17 update

**Comment by xabbuh** [comment]
@mdeherder I think your issue was fixed by #39241.

**Comment by mdeherder** [comment]
I tested with this code and the problem persists :
```php
        try {
            if ('' !== $value && '{' === $value[0]) {
                // $cursor = \strlen($this->currentLine) - \strlen($value);
                $cursor = \strlen(rtrim($this->currentLine)) - \strlen(rtrim($value));

                return Inline::parse($this->lexInlineMapping($cursor), $flags, $this->refs);
            } elseif ('' !== $value && '[' === $value[0]) {
                // $cursor = \strlen($this->currentLine) - \strlen($value);
                $cursor = \strlen(rtrim($this->currentLine)) - \strlen(rtrim($value));

                return Inline::parse($this->lexInlineSequence($cursor), $flags, $this->refs);
            }
```

**Comment by xabbuh** [comment]
@mdeherder Can you check #39274?

**Comment by mdeherder** [comment]
@xabbuh 
Ok with #39274

**Comment by Nyholm** [comment]
Awesome. Closing this a a duplicate of #39229

**Comment by Nyholm** [comment]
Ooops. Sorry. I was too quick. This **will** be fixed by #39274

**Comment by xabbuh** [comment]
Actually, #39267 will fix the initial report. #39274 fixed another regression.
